xc7/utils/prjxray_create_edges.py

```diff
@@ -1031,39 +1046,37 @@ def main():
 
         edge_set = set()
 
-        for loc in progressbar.progressbar(grid.tile_locations()):
-            gridinfo = grid.gridinfo_at_loc(loc)
-            tile_name = grid.tilename_at_loc(loc)
+        c2 = conn.cursor()
+        for tile_pkey, tile_type_pkey, grid_x, grid_y in \
+                progressbar.progressbar(
+                c.execute(
+                    "SELECT pkey, tile_type_pkey, grid_x, grid_y FROM tile")):
+            loc = (grid_x, grid_y)
 
             # Not a synth node, check if in ROI.
-            if use_roi and not roi.tile_in_roi(loc):
+            if use_roi and not (
+                    roi_range[0] <= loc[0] and loc[0] <= roi_range[2]
+                    and roi_range[1] <= loc[1] and loc[1] <= roi_range[3]):
                 continue
 
-            tile_type = db.get_tile_type(gridinfo.tile_type)
+            # Process PIPs
+            for pip_pkey, pip_name, pip_src_wire_pkey, pip_dst_wire_pkey in\
+                    c2.execute("""SELECT pkey, name, src_wire_in_tile_pkey,
+                               dest_wire_in_tile_pkey FROM pip_in_tile WHERE
+                               tile_type_pkey = (?)""",
+                (tile_type_pkey, )):
 
-            for pip in tile_type.get_pips():
-                if pip.is_pseudo:
-                    continue
-
-                if not pip.is_directional:
-                    # TODO: Handle bidirectional pips?
-                    continue
+                # No pseudo pips and bidirectional pips should be present here
+                # as they were skipped during import to the SQLite database
+                # in prjxray_form_channels.py
 
                 connections = make_connection(
-                    conn=conn,
-                    input_only_nodes=input_only_nodes,
-                    output_only_nodes=output_only_nodes,
-                    find_pip=find_pip,
-                    find_wire=find_wire,
-                    find_connector=find_connector,
-                    tile_name=tile_name,
-                    loc=loc,
-                    tile_type=gridinfo.tile_type,
-                    pip=pip,
-                    switch_pkey=switch_pkey,
-                    delayless_switch_pkey=delayless_switch_pkey,
-                    const_connectors=const_connectors
+                    conn, input_only_nodes, output_only_nodes, loc, tile_pkey,
+                    pip_src_wire_pkey, pip_dst_wire_pkey, pip_pkey,
+                    switch_pkey, delayless_switch_pkey, find_connector,
+                    const_connectors
```

Comment:
Please continue to use kwarg calling style when there are this many arguments.

Reply:
Done.